### PR TITLE
[HIGH] Feature/click document

### PR DESCRIPTION
**Pull request by mugnivenko**
click document https://app.clickup.com/t/1nbqet2

care charges reset https://app.clickup.com/t/1p2jhkk

image

### CODE REVIEW

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
Screenshot 2021-12-08 at 09 11 48

File: components/ViewDocument.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  const mainLoading = isLoading || localLoading;
+
+  const setMainLoading = setIsLoading || setLocalLoading;
+
+  if (!hasFile) return <p>N/A</p>;
+
+  return useMemo(() => (
+    <Button
+      link=''
```
Comment (mugnivenko) [inline components/ViewDocument.js:24]: added link='' to fix boolean issue

Comment (mugnivenko) [reply components/ViewDocument.js]: also i removed useMemo, there is no reason for it

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
This should be a green download button
Screenshot 2021-12-09 at 10 50 16
then when you press it, it should download the file and then show a 'mark as paid button'... it doesn't show the mark as paid button unless you manually refresh the page

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
cannot see FNC file on review care package page even though i have uploaded one?

**Comment by mugnivenko** [comment]
> cannot see FNC file on review care package page even though i have uploaded one?

yep, BE didn't send file info. Wait for Furkan fix it

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
merge issues

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
once you have downloaded the file, clicking the download button again should open the originally downloaded version rather than calling the endpoint again (e.g. cache/save the blob inside the component and reuse it rather than re-downloading it)

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
/payruns/ec2cfc82-4c2a-4ef7-aaf0-ddd01bbd555f

pressing download should not refresh page

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
get an error trying to upload file here: care-package/0612c38f-354d-4450-b59a-f3a8ed4412b7/core-package
Screenshot 2021-12-17 at 13 40 37
Screenshot 2021-12-17 at 13 41 06

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
file upload blows up on care charge page

Screenshot 2021-12-17 at 13 45 17

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
doesnt work /broker-referral/4/assign-package either
i don't think file upload works anywhere at the moment on this branch

**Comment by mugnivenko** [comment]
1) I fixed getFormData function, there is no need to pass object. 

2) One more problem with uploading file using PUT request. I think it's BE side issue.

**Review by akeamwoodsND — CHANGES_REQUESTED** [review]
file upload gets this error on broker-referral/GUID/assign-package but **not on develop**
Screenshot 2021-12-20 at 13 02 40

**Comment by mugnivenko** [comment]
> file upload gets this error on broker-referral/GUID/assign-package but **not on develop** Screenshot 2021-12-20 at 13 02 40

this is BE issue. Furkan working on it

**Pull request merged**